python/StartServer.py

```diff
@@ -15,18 +16,76 @@ class SpacetimeConsole(cmd.Cmd):
     prompt = 'Spacetime> '
 
     """Command console interpreter for frame."""
+    def do_quit(self, line):
+        """ quit
+        Exits all applications by calling their shutdown methods.
+        """
+        shutdown()
+
     def do_exit(self, line):
         """ exit
         Exits all applications by calling their shutdown methods.
         """
         shutdown()
 
+    def do_objsin(self, type_text):
+        """ objsin <type>
+        Prints the primary key of all objects of a type (accepts auto-complete)
+        """
+        objs = fs.Store.get(fs.name2class[type_text])
+        if objs:
+            print "{0:20s}".format("ids")
+            print "============="
+            for oid in objs:
+                print "{0:20s}".format(oid)
+            print ""
+
+    def complete_objsin(self, text, line, begidx, endidx):
+        if not text:
+            completions = [t.Class().__name__ for t in DATAMODEL_TYPES]
+        else:
+            completions = [t.Class().__name__ for t in DATAMODEL_TYPES if t.Class().__name__.startswith(text)]
+        return completions
+
+    def complete_list(self, text, line, begidx, endidx):
+        return ['sets','apps']
+
+    def do_list(self, line):
+        """ list ['sets','apps']
+        list accepts one of two arguments:
+        * 'sets' prints all pcc sets tracked by the server
+        * 'apps' prints the name of all applications registered with the server
+        """
+        if line == "sets":
+            for t in DATAMODEL_TYPES:
+                print "{0:60s}{1:s}".format(t.Class().__name__, t.Class().__module__)
+        elif line == "apps":
+            all_apps = fs.Store.get_app_list()
+            for app in all_apps:
+                print app
+        else:
+            print line
+
+    def do_clear(self, type_text):
+        """ clear <type>
+        Deletes all objects.
+
+        If a type is passed, deletes all objects of that type.
+        """
+        if type_text:
+            fs.Store.clear(fs.name2class[type_text])
+        else:
+            fs.Store.clear()
```

Comment:
better to be explicit like:

clear all
or something.
so that if someone is typing clear and then by mistake presses enter, all objects are not deleted.